tests/Microsoft.Bot.Builder.TestBot/Startup.cs

```diff
@@ -1,93 +1,39 @@
 
-using System;
-using System.Linq;
-using System.Threading.Tasks;
 using Microsoft.AspNetCore.Builder;
 using Microsoft.AspNetCore.Hosting;
-using Microsoft.AspNetCore.Http;
-using Microsoft.Bot.Builder.Dialogs;
-using Microsoft.Bot.Builder.Integration;
+using Microsoft.AspNetCore.Mvc;
 using Microsoft.Bot.Builder.Integration.AspNet.Core;
+using Microsoft.Bot.Builder.TestBot.Bots;
+using Microsoft.Bot.Connector.Authentication;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
-using Microsoft.Extensions.Options;
 
 namespace Microsoft.Bot.Builder.TestBot
 {
     public class Startup
     {
-        public Startup(IHostingEnvironment env)
+        public Startup(IConfiguration configuration)
         {
-            var builder = new ConfigurationBuilder()
-                .SetBasePath(env.ContentRootPath)
-                .AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
-                .AddJsonFile($"appsettings.{env.EnvironmentName}.json", optional: true)
-                .AddEnvironmentVariables();
-
-            Configuration = builder.Build();
+            Configuration = configuration;
         }
 
         public IConfiguration Configuration { get; }
 
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
-            //services.AddSingleton<IAdapterIntegration>(sp =>
-            //{
-            //    var options = sp.GetRequiredService<IOptions<BotFrameworkOptions>>().Value;
-            //    var accessors = sp.GetRequiredService<TestBotAccessors>();
-
-            //    options.Middleware.Add(new AutoSaveStateMiddleware(accessors.ConversationState));
-            //    options.Middleware.Add(new ShowTypingMiddleware());
-
-            //    var botFrameworkAdapter = new BotFrameworkAdapter(options.CredentialProvider, options.ChannelProvider, options.ConnectorClientRetryPolicy, options.HttpClient)
-            //    {
-            //        OnTurnError = options.OnTurnError,
-            //    };
-
-            //    foreach (var middleware in options.Middleware)
-            //    {
-            //        botFrameworkAdapter.Use(middleware);
-            //    }
-
-            //    //return botFrameworkAdapter;
-
-            //    return new InteceptorAdapter(botFrameworkAdapter);
-            //});
+            services.AddMvc().SetCompatibilityVersion(CompatibilityVersion.Version_2_1);
 
-            IStorage dataStore = new MemoryStorage();
-            var conversationState = new ConversationState(dataStore);
+            // Load the credentials from configuration and create the credential provider.
+            var appId = Configuration["BotFramework:AppId"];
+            var password = Configuration["BotFramework:Password"];
+            var credentialProvider = new SimpleCredentialProvider(appId, password);
```

Comment:
This makes me laugh. 

Can we update the SipleCredentialProvider so this "glue" code is not necessary.

Reply:
you mean have the constructor take the dictionary as argument?